### PR TITLE
[WIP] Matrix synapse

### PR DESCRIPTION
**Pull request by NoctuaNivalis**
This still ought to be tested and checked and cleaned.

### CODE REVIEW

File: site.yml [hunk 1]
```diff
@@ -44,5 +44,6 @@
 #    - django
 #    - user_configs
 #    - dokuwiki
+#    - matrix
```
Comment (FlashYoshi) [inline site.yml]: My newline :(

**Comment by JensTimmerman** [comment]
the default homeserver.yaml is not exactly what you want, 
maybe we could ship one that you generate and edit yourself?

e.g. you want to change the webport? you want to turn on registration, and you want to switch from sqlite db to a real database?

**Comment by JensTimmerman** [comment]
and maybe turn on captcha's for now?

**Comment by NoctuaNivalis** [comment]
I've kept registeration off, but allowed registration using a "secret" key. It's in the pushed config file.

**Comment by redfast00** [comment]
We probably won't use this since we'll be using Rocket.Chat

**Pull request closed without merging**